pkg/driver/controller.go

```diff
@@ -265,19 +265,25 @@ func (cs *controller) DeleteSnapshot(
 	ctx context.Context,
 	req *csi.DeleteSnapshotRequest,
 ) (*csi.DeleteSnapshotResponse, error) {
+
 	snapshotID := strings.Split(req.SnapshotId, "@")
+	// In case of migrated volume snapshot the snapshotID will be of the form:
+	// pv-name+"_"+snapshot_name.
+	if len(snapshotID) == 1 {
+		snapshotID = strings.SplitN(req.SnapshotId, "_", 2)
+	}
 	if len(snapshotID) != 2 {
 		return nil, status.Errorf(
 			codes.Internal,
-			"failed to handle CreateSnapshotRequest for %s, {%s}",
+			"failed to handle DeleteSnapshotRequest for %s, {%s}",
 			req.SnapshotId,
 			"Manual intervention required",
 		)
 	}
 	if err := utils.DeleteSnapshot(snapshotID[0], snapshotID[1]); err != nil {
```

Comment:
In case of external-provisioned snapshot `snapshotID[1]` should be the entire string i.e. req.SnapshotId.

Reply:
Closing the PR , as we are trying to populate the snapshotID which is honoured in CSI-Driver , which not required any code changes